src/test/url-handling.test.js

```diff
@@ -189,6 +190,41 @@ describe('url handling tracks', function() {
         '  - hide [thread Style]',
       ]);
     });
+
+    it('can select a local track without mixing track and thread indexes', function() {
+      // We're building a very specific profile, where local track indexes and
+      // thread indexes could be confused. This is easier if we have local
+      // tracks for the first process, because then the thread indexes and the
+      // local track indexes are off by one.
```

Comment:
Can you include the GitHub issue number where this happened? This is a very specific regression fix, and not a largely generalizable problem that we're testing, so it would be nice to have the history of the specific issue mentioned in the test to have a bit more context.

Also, it would be helpful to mention that the selected track is index `0` and the selected thread is index `1`. It took me awhile to realize that was what was being asserted here.

Reply:
Basically the assertion checks that when computing the selected thread, "111-1" doesn't remove the 2nd thread (we want to select), but the 2nd track.

This is a bit confusing in the code that `getVisibleThreads` is actually used only for this, and not to actually hide stuff :-)